pressbooks/pressbooks
# feat: support edited volumes

This allows the publishing of Edited Volumes and other books that don't have a book level author, as well as implementing the other parameters that @SteelWagstaff outlined here [idea/issue134](https://github.com/pressbooks/ideas/issues/134#issuecomment-944830722). 

This pull request works in conjunction with the corresponding [pressbooks-book pull request](https://github.com/pressbooks/pressbooks-book/pull/981). 


### What it does
1. Allows users to enter book meta data without authors, so long as there is an editor. If there is no editor and no author, defaults to the same behavior (using the book owner).
2. Displays "Edited By <Editor Name>" in place of the Author on title pages when no Author is present. 
3. Displays "Edited By <Editor Name>" below the Author Name when <Editor_Name> is listed as the book editor and there is an author. Same pattern for Translators and Illustrators
4. Makes "Book Editor" available as a running head/foot element 
5. If Contributors are present, adds "Contributors " before the list of contributors. 

### How to Test
First, checkout the related pressbooks-book [feature branch](https://github.com/UIUCLibrary/pressbooks-book/tree/issue_134) from this [pull request](https://github.com/pressbooks/pressbooks-book/pull/981)
1. Allows users to enter book meta data without authors, so long as there is an editor. If there is no editor and no author, defaults to the same behavior (using the book owner).

#### To test: 
1a. From Book Info, add some editors, remove all authors and save. Changes should be preserved. Result:
<img width="1400" alt="Screen Shot 2022-02-22 at 4 33 43 PM" src="https://user-images.githubusercontent.com/47127862/155223437-96195ef5-f72b-41a2-8c65-3dd8fc3197ee.png">
1b Next, Remove all editors and save. Book owner should be added as author. Result:
<img width="1400" alt="Screen Shot 2022-02-22 at 4 36 51 PM" src="https://user-images.githubusercontent.com/47127862/155223747-e4804ac7-a7ff-4459-b7bd-2d3389eb8f9d.png">

2. Displays "Edited By <Editor Name>" in place of the Author on title pages when no Author is present. 

#### To test:  Repeat 1a and click "Visit Book". Books should display "Edited By <Editor Name>" where an author's name would normally appear. Result:
<img width="1396" alt="Screen Shot 2022-02-22 at 4 45 49 PM" src="https://user-images.githubusercontent.com/47127862/155224949-d14f3de1-9f61-4eab-99a5-0f2cbbb035b3.png">
2.1.b Export book to epub, pdf, print pdf, html book. Same patter as above should appear. Results:

epub
<img width="547" alt="Screen Shot 2022-02-22 at 4 52 31 PM" src="https://user-images.githubusercontent.com/47127862/155225781-b0a68b1c-01d3-484b-ad57-6123fe78ca58.png">

pdf:
<img width="868" alt="Screen Shot 2022-02-22 at 4 51 25 PM" src="https://user-images.githubusercontent.com/47127862/155225633-1fcb592e-4666-4fef-9d30-b074cd660c24.png">

print pdf:
same as pdf

html book:
<img width="1399" alt="Screen Shot 2022-02-22 at 4 48 29 PM" src="https://user-images.githubusercontent.com/47127862/155225304-89f4b36e-2b81-4e1e-ae19-2f0416a59db7.png">


#### 3. Displays "Edited By <Editor Name>" below author when <Editor_Name> is listed as the book editor and there is an author. Same pattern for Translators and Illustrators.

#### To test: Same as 2 above, but add an author. 
pdf:
<img width="354" alt="Screen Shot 2022-02-22 at 5 06 50 PM" src="https://user-images.githubusercontent.com/47127862/155227667-258eac5e-c1dc-417c-8ac9-f233d054b39f.png">
ebpub:
<img width="507" alt="Screen Shot 2022-02-22 at 5 07 44 PM" src="https://user-images.githubusercontent.com/47127862/155227723-7fdaa162-5c29-482c-8e84-114b1575d037.png">
HTML book:
<img width="507" alt="Screen Shot 2022-02-22 at 5 07 44 PM" src="https://user-images.githubusercontent.com/47127862/155227798-749a7090-267f-49fb-a74c-a65b38e503e0.png">

#### 4. Makes "Book Editor" available as a running head/foot element 

#### To test: From Theme Options>PDF Options> Running Heads and Feet select an option that will display in your book and theme and then export. Result:
<img width="1433" alt="Screen Shot 2022-02-22 at 5 38 14 PM" src="https://user-images.githubusercontent.com/47127862/155231423-3d06f4b7-9c28-471b-9437-5c4c1a7f9325.png">


#### 5. If Contributors are present, adds "Contributors: " before the list of contributors. 

#### To test: Add contributors and export. Result (epub shown):
<img width="521" alt="Screen Shot 2022-02-22 at 5 41 26 PM" src="https://user-images.githubusercontent.com/47127862/155231832-87cfa438-aae9-455f-9874-3db38325b6c0.png">
